README.md

```diff
@@ -78,14 +78,48 @@ If you want to link to a page that exists on https://pulumi.com but not in this
 
 ### Temporarily pulling in content from pulumi/docs
 
-If the change you're working on requires content from pulumi/docs -- e.g., the aforementioned how-to guides or package documentation -- you may want to be able to see some of that content as you develop. To do that, just copy the files you need from the [`content` folder of pulumi/docs](https://github.com/pulumi/docs/tree/master/content) into the `content` folder of this repository (and/or the [`static` folder](https://github.com/pulumi/docs/tree/master/content), if you're working on API Docs changes), if you're looking for the JSON files that power the API Docs navigation), remembering to remove those files before you commit. For example:
+If the change you're working on requires content from pulumi/docs -- e.g., the aforementioned how-to guides -- you may
+want to be able to see some of that content as you develop. To do that, just copy the files you need from the
+[`content` folder of pulumi/docs](https://github.com/pulumi/docs/tree/master/content) into the `content` folder of this
+repository, remembering to remove those files before you commit. For example:
 
 ```
-# Copy the AWS API docs and navigation from a local/sibling clone of pulumi/docs.
-cp -R ../docs/content/registry/packages/aws/api-docs ./themes/default/content/registry/packages/aws/
-cp ../docs/static/registry/packages/navs/aws.json ./themes/default/static/registry/packages/navs/
+# Copy the AWS how-to guides from a local/sibling clone of pulumi/docs.
+cp ../docs/content/registry/packages/aws/how-to-guides ./themes/default/content/registry/packages/aws/
 ```
 
+#### Temporarily generating API docs for packages
+
+The API docs for packages can be generated on-demand using the `resourcedocsgen` tool. The PR build for this repo, for instance,
+generates the API docs for all packages from the latest commit hash of the PR being built. Likewise, as long as your branch
+is pushed up to GH, you can generate API docs using any commit hash from your PR branch locally as well. This is because
```

Comment:
> but I suspect most users who've cloned the registry to work on a branch would be happy just to generate docs with that same branch (assuming this is even possible -- which now that I write this, I realize it may not be?)

hmm I don't quite follow the question here. Are you asking if it is possible to generate API docs using a specific branch of the registry? If yes, then yes. It's possible with the `--branch` flag. If you omit the `--branch` flag, then `master` branch will be the default.

Reply:
I updated this based on what I think I understood. Not sure if this is now a bit more clear to read. Can you please check the latest and let me know what you think?

Reply:
Ok yeah, sorry for my confusing words! What I meant was the most common use case is probably something like this:

1. Clone master and make a feature branch
1. Start developing
1. Notice you don't have the API docs you want
1. Build them

So the README should probably lead with making it clear how to do just that. (Which I believe would look like the example I quoted -- since what `--branch` would they be providing if they haven't pushed a branch yet?) The more advanced use case woudl then be "I want to pull API docs from somewhere _other_ than master", in which case we can explain how to do that by showing an example with `--branch` or `commitSha`. Does that make sense?